config/initializers/content_security_policy.rb

```diff
@@ -2,17 +2,34 @@
 # For further information see the following documentation
 # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
 
-# Rails.application.config.content_security_policy do |p|
-#   p.default_src :self, :https
-#   p.font_src    :self, :https, :data
-#   p.img_src     :self, :https, :data
-#   p.object_src  :none
-#   p.script_src  :self, :https
-#   p.style_src   :self, :https, :unsafe_inline
-#
-#   # Specify URI for violation reports
-#   # p.report_uri "/csp-violation-report-endpoint"
-# end
+if Rails.env.production?
+  assets_host = Rails.configuration.action_controller.asset_host || "https://#{ENV['WEB_DOMAIN'] || ENV['LOCAL_DOMAIN']}"
+  data_hosts = [assets_host]
+
+  if ENV['S3_ENABLED'] == 'true'
+    attachments_host = ENV['S3_ALIAS_HOST'] || ENV['S3_CLOUDFRONT_HOST'] || ENV['S3_HOSTNAME'] || "s3-#{ENV['S3_REGION'] || 'us-east-1'}.amazonaws.com"
+  elsif ENV['SWIFT_ENABLED'] == 'true'
+    attachments_host = ENV['SWIFT_OBJECT_URL']
+  else
+    attachments_host = nil
+  end
+  data_hosts << attachments_host unless attachments_host.nil?
+
+  Rails.application.config.content_security_policy do |p|
+    p.base_uri        :none
+    p.default_src     :none
+    p.frame_ancestors :none
+    p.script_src      :self, assets_host
+    p.font_src        :self, assets_host
+    p.img_src         :self, :https, :data, :blob
+    p.style_src       :self, :unsafe_inline, assets_host
+    p.media_src       :self, :data, *data_hosts
+    p.frame_src       :self, :https
+    p.worker_src      :self, assets_host
+    p.connect_src     :self, :blob, Rails.configuration.x.streaming_api_base_url, *data_hosts
```

Comment:
Adding the `:blob` to connect_src resolves another error I was running into after previous CSP changes. The error was like:

`Content Security Policy: The page’s settings blocked the loading of a resource at blob:https://$my_instance/$redacted_url (“connect-src”).`

I was seeing this when trying to upload a JPG attachment. So 👍 on the `:blob` addition.

Reply:
I do not believe that connect-src should affect image resizing. That should be img-src...

Reply:
But it does. I don't remember why exactly, it might be the `readAsDataURL` or `canvas.toBlob`.